Sources/SwiftDependencyGraphs/DependencyGraph.swift

```diff
@@ -132,6 +132,55 @@ public struct DependencyGraph<V> where V: Hashable, V: Identifiable, V: Sendable
     return forwards.union(backwards)
   }
 
+  public typealias RemoveVertexResult = Result<RemoveVertexSuccess<V>, RemoveVertexError<V>>
+
+  /// Removes a vertex from the graph.
+  /// - Parameters:
+  ///   - vertex: The removed vertex.
+  ///   - isForced: `true` to also remove edges to other vertices if there are any.
+  ///   `false` to remove `vertex` iff it has no edges to other vertices.
+  /// - Returns: The result of the removal. If successful, the removed vertex and edges are returned.
+  /// Otherwise, the reason for the failure is returned.
+  @discardableResult mutating public func remove(
+    vertex: V, byForce isForced: Bool = false
+  ) -> RemoveVertexResult {
+    guard vertices[vertex.id] != nil else {
+      return .failure(RemoveVertexError.notInGraph(vertex))
+    }
+
+    if !isForced {
+      guard let outgoingEdges = outgoingEdges[vertex.id],
+        outgoingEdges.isEmpty,
+        let incomingEdges = incomingEdges[vertex.id],
+        incomingEdges.isEmpty
+      else {
+        return .failure(
+          RemoveVertexError.hasEdgesTo(
```

Comment:
With strong typing for the result, you should be able to omit the type name here (which Swift devs usually seem to like when otherwise it gets repetitive with types that just have technical reasons to exist, opposed to e.g. expressing something in the domain)

```suggestion
         .hasEdgesTo(
```

Reply:
Done.